src/app/routes/applications/new-role/new-role.component.ts

```diff
@@ -388,6 +393,24 @@ export class NewRoleComponent implements OnInit, AfterViewInit, OnDestroy {
     }
   }
 
+  addRestriction(event: ISmartSearch) {
+    if (event.SearchType === 'restrictions') {
+      if (this.roleForm.get('data').value.enrolmentPreconditions[0].conditions.length > 0) {
```

Comment:
This part is wrong. We should be able to put as many precondition as we want. Task title is: `Role creation should receive an array of roles preconditions`

Reply:
In the description, I saw `Now, only one precondition can be set.`, so that's why I added that condition.

Reply:
Yeah, important word `now` :D Right now, i mean without any changes we can set one precondition. We want to set them more. This task is about extending this functionality, so we can add multiple preconditions.